src/components/career-form/career-file-upload.tsx

```diff
@@ -0,0 +1,238 @@
+import { Input, InputContainer } from '../form/controls';
+import React, { ReactNode, useCallback } from 'react';
+import { DropzoneRootProps, useDropzone } from 'react-dropzone';
+import styled from 'styled-components';
+import { theme } from '../layout/theme';
+import {
+  FieldErrors,
+  FieldValues,
+  UseFormClearErrors,
+  UseFormRegister,
+  UseFormSetError,
+  UseFormSetValue,
+  UseFormWatch,
+} from 'react-hook-form';
+import { GridItem } from '../grid/grid';
+import { FilePreview } from './career-file-preview';
+import { FormError } from './career-components';
+import { CareerSelect } from './career-select';
+import { FileTitleWrapper } from './career-file-title';
+
+interface FileUploadProps {
+  setValue: UseFormSetValue<FieldValues>;
+  setError: UseFormSetError<FieldValues>;
+  clearErrors: UseFormClearErrors<FieldValues>;
+  register: UseFormRegister<FieldValues>;
+  watch: UseFormWatch<FieldValues>;
+  name: string;
+  selectedFiles: FileList;
+  errors: FieldErrors;
+  children: ReactNode | ReactNode[];
+}
+
+const MAX_NUMBER = 3;
+const MAX_SIZE = 20000000; // maximal size per document of personio
+
+export const FileUpload = ({
+  setValue,
+  setError,
+  clearErrors,
+  name,
+  register,
+  selectedFiles,
+  errors,
+  children,
+  watch,
+}: FileUploadProps) => {
+  const onDrop = useCallback(
+    (droppedFiles) => {
+      // maximal number of file being dropped
+      const maximalOfFiles = selectedFiles
+        ? MAX_NUMBER - selectedFiles.length
+        : MAX_NUMBER;
+      // dropped files are appended to already selected files
+      const dataTransfer = new DataTransfer();
+      if (selectedFiles) {
+        for (let i = 0; i < selectedFiles.length; i++) {
+          dataTransfer.items.add(selectedFiles[i]);
+        }
+      }
+      for (let i = 0; i < Math.min(maximalOfFiles, droppedFiles.length); i++) {
```

Comment:
good knowledge: the part `Math.min(maximalOfFiles, droppedFiles.length);` is never changing but by writing it into the loop it's being evaluated as often the loop is running. Instead you could initialize the limit in the first part of the loop (where `i=0` is set) as it's run only once. Super tiny detail which is not worth to optimize here though.

Reply:
You calculate that value ` Math.min(maximalOfFiles, droppedFiles.length)` at least two times in this component so you can also just calculate once and store it at the beginning too